rfd/0217-bot-details-web.md

```diff
@@ -0,0 +1,280 @@
+
+![](assets/0217-feature-delete-bot.png)
+#### Lock bot
+Locks the bot after confirmation. Shows a loading indicator during the call to the api. On success, removes the dialog but remains on the bot detail page. On error, shows a message within the confirmation dialog.
+
+![](assets/0217-feature-lock-bot.png)
+
+Once a bot is locked, to unlock it requires navigating to **Identity Governance > Session & Identity Locks**, finding the lock in question and removing it. Deleting a lock requires it's UUID, which wont be know on this page.
+### Implementation
+
+#### Data fetching and caching
+
+In order to keep the implementation modular, each logical section of the page fetches its own data, shows loading status and manages error states. This increases flexibility for future iterations while reducing the likelihood of introducing regression issues. Data caching behaviour can be tailored to each area of data independently.
+
+#### Web APIs
+
+##### `GET /v1/webapi/sites/:site/machine-id/bot/:name`
+
+Fetch a bot by name, including roles and traits. Endpoint exists and will be reused. Additional fields for lock state will be added to the Bot resource and populated from the underlying User resource.
+
+``` protobuf
+message BotStatus {
+  // IsLocked indicates if the bot user is locked
+  bool IsLocked = 4
+  // LockedMessage contains the message to display when the bot user is locked
+  string LockedMessage = 5
+  // LockedTime contains the time when bot user was locked
+  google.protobuf.Timestamp LockedTime = 6
+  // LockExpires contains the time this lock will expire
+  google.protobuf.Timestamp LockExpires = 7
+}
+```
+
+##### `GET /v1/webapi/tokens?filter_role=Bot&filter_bot_name=:name`
+
+Fetch join tokens linked to a bot by name.
+
+**Approach**
+Existing endpoint with additional filters added. Filter params will be added; `filter_role` and `filter_bot_name`
+
+``` protobuf
+// GetTokensRequest is used to request a list of tokens. Filtering by role and bot_name are supported
+message GetTokensRequest {
+    // FilterRole allows filtering the returned items by the token's role (e.g. 'static', 'node', 'bot', etc)
+    string FilterRole = 1 [(gogoproto.jsontag) = "filter_role"];
+    // FilterBotName allows filtering the returned items by the name of the associated bot. This is a no-op unless FilterRole is 'bot'.
+    string FilterBotName = 2 [(gogoproto.jsontag) = "filter_bot_name"];
+}
+
+service AuthService {
+  // GetTokens retrieves all tokens.
+  rpc GetTokens(GetTokensRequest) returns (types.ProvisionTokenV2List);
+}
+```
+_api/proto/teleport/legacy/client/proto/authservice.proto_
+
+**Performance**
+The existing endpoint does not support pagination. It is not anticipated that use of the endpoint for the Bot Details page will increase usage or load significantly, as such, the endpoint will be used as-is. A bot is likely to have fewer than 20 associated join methods.
```

Comment:
If we're making a new RPC maybe we should just support pagination even though we don't have an immediate need for it. We've done it enough times by now that there should be plenty of examples to copy from - I wouldn't expect it to create much additional work for you.